scripts/copy_qubit.py

```diff
@@ -72,9 +72,16 @@ def get_data(file_path):
                 read=True
             elif(read and row[sample_index]):
                 #this is every other row
-                data[row[sample_index]]={}
-                data[row[sample_index]]['concentration']=row[conc_index]
-                data[row[sample_index]]['unit']=row[unit_index]
+                if row[sample_index] in data:
+                    #Sample is duplicated, drop the key
+                    log.append("sample {0} has two rows in the Qubit CSV file. Please check the file manually.".format(row[sample_index]))
+                    del data[row[sample_index]]
```

Comment:
I could stop the whole script, but that would mean that the others fields
would not be updated properly, so I would rather update everything so they
can check the errors all at once instead of having to run the EPP and
gettting stuck each time.

On Fri, Mar 27, 2015 at 4:16 PM, senthil10 <notifications@github.com> wrote:

> In scripts/copy_qubit.py
> <https://github.com/SciLifeLab/genologics/pull/100#discussion_r27302174>:
>
> > @@ -72,9 +72,16 @@ def get_data(file_path):
> >                  read=True
> >              elif(read and row[sample_index]):
> >                  #this is every other row
> > -                data[row[sample_index]]={}
> > -                data[row[sample_index]]['concentration']=row[conc_index]
> > -                data[row[sample_index]]['unit']=row[unit_index]
> > +                if row[sample_index] in data:
> > +                    #Sample is duplicated, drop the key
> > +                    log.append("sample {0} has two rows in the Qubit CSV file. Please check the file manually.".format(row[sample_index]))
> > +                    del data[row[sample_index]]
>
> Ah ha, I see. If they (technician) are going/have to change the file and
> re run it again, cant the script just throw an error to do so and stop
> executing rather than to continue ?
>
> The loop is fine as the way it is, I just got confused as you remove the
> key form data [image: :sweat_smile:]
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/SciLifeLab/genologics/pull/100/files#r27302174>.
>